src/dynsight/_internal/analysis/spatial_average.py

```diff
@@ -117,6 +117,19 @@ def spatialaverage(
             If the input descriptor array does not have 2 or 3 dimensions,
             an error is raised.
 
+    Warning:
```

Comment:
I would move this above `Key features`

Additionally, I would change `key features` for `.. important::` (from https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/#notes)